addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/AbstractInputControlXML.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.yamahareceiver.internal.protocol.xml;
+
+import org.openhab.binding.yamahareceiver.internal.protocol.AbstractConnection;
+
+import java.lang.ref.WeakReference;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.openhab.binding.yamahareceiver.YamahaReceiverBindingConstants.INPUT_MUSIC_CAST_LINK;
+import static org.openhab.binding.yamahareceiver.YamahaReceiverBindingConstants.INPUT_NET_RADIO;
+import static org.openhab.binding.yamahareceiver.YamahaReceiverBindingConstants.INPUT_TUNER;
+
+/**
+ * Provides basis for all input controls
+ *
+ * @author Tomasz Maruszak - Initial contribution
+ */
+public abstract class AbstractInputControlXML {
+
+    protected final WeakReference<AbstractConnection> comReference;
+    protected final String inputID;
+    protected final Map<String, String> inputToElement;
+
+    private Map<String, String> loadMapping() {
+        Map<String, String> map = new HashMap<>();
+
+        // ToDo: For maximum compatibility these should be obtained fro the VNC_Tag of the desc.xml
```

Comment:
I left this ToDo for future, so that if still something is not compatible on some Yamaha models this would be another area to enhance. For now, I don't plan to leverage VNT_Tag as everything seems to work according to users.
If you prefer, I can take out this ToDo line, but I feel it's better to have.

Reply:
No its, fine just a check